oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java

```diff
@@ -1306,9 +1306,13 @@ private void updateCoords(String cluster, BundleJob bundle,
                 delay = getDelay((Feed) entity, coord);
             }
 
-            //calculate next start time based on delay.
-            endTime = (delay == null) ? endTime
-                    : EntityUtil.getNextStartTime(coord.getStartTime(), delay, EntityUtil.getTimeZone(entity), endTime);
+            //calculate end time based on delay only for feed with delay.
```

Comment:
Just for reference. https://github.com/InMobi/falcon/blob/master/oozie/src/main/java/org/apache/falcon/oozie/feed/FeedReplicationCoordinatorBuilder.java#L105
The delay gets added to the start time of the new coordinator.